aidb/query/query.py

```diff
@@ -478,6 +484,41 @@ def is_valid_aqp_query(self):
     return True
 
 
+  @cached_property
+  def recall_target(self):
+    recall_target = self._get_keyword_arg(exp.RecallTarget)
+    return recall_target / 100. if recall_target else None
+
+
+  @cached_property
+  def precision_target(self):
+    precision_target = self._get_keyword_arg(exp.PrecisionTarget)
+    return precision_target / 100. if precision_target else None
+
+
+  @cached_property
+  def oracle_bughet(self):
+    oracle_budget = self._get_keyword_arg(exp.Budget)
+    return oracle_budget if oracle_budget else None
+
+
+  @cached_property
+  def is_approx_select_query(self):
+    if self.precision_target is not None:
+      raise Exception("We haven't support approx select query with precision target.")
+    return self.recall_target is not None
+
+
+  def is_valid_approx_select_query(self):
+    if self.oracle_bughet:
```

Comment:
Functions that are "is_" should return True or False and the caller should raise based on the result